staging/src/kubevirt.io/client-go/api/v1/types.go

```diff
@@ -1215,6 +1215,9 @@ const (
 	// VirtualMachineStatusUnknown indicates that the state of the virtual machine could not be obtained,
 	// typically due to an error in communicating with the host on which it's running.
 	VirtualMachineStatusUnknown VirtualMachinePrintableStatus = "Unknown"
+	// VirtualMachineStatusUnschedulable indicates that an error has occurred while scheduling the virtual machime,
+	// e.g. due to unsatisfiable resource requests or unsatisfiable scheduling constraints.
+	VirtualMachineStatusUnschedulable VirtualMachinePrintableStatus = "FailedUnschedulable"
```

Comment:
Maybe simple `Unschedulable`? or `FailedScheduling`?

Reply:
I think that all of these options are viable.

- The `Failed` prefix somewhat implies that this failure has occurred while attempting to do something (starting the VM). I think that `Unschedulable` somewhat implies that this is more of a universal state that can always be discovered, whereas in practice we can only discover it while the virt-launcher pod is created and fails to be scheduled.
- The virt-launcher pod fires an event with reason `FailedScheduling`. The VMI sets the `PodScheduled=False` condition with reason `Unschedulable`. I kinda mixed these two to form the `FailedUnschedulable`.

In summary I'm open to all of these, don't have a strong opinion towards any of them. If someone else does, I'm good with changing accordingly :-)